### DISCUSSION TITLE
Query returning unchanged components

### DISCUSSION THREAD

**Discussion by s-mayrh** [bevyengine/bevy#9563] [Q&A] [answered]
Hi! I have a very basic Bevy application with a single system that queries for a specific component and prints some information when it's updated. However, even though there are no changes to the queried component, the system seems to contain a query result every frame and prints the output.

Here's the simplified version of my code:
```
use bevy::prelude::*;

fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .add_startup_system(spawn_player)
        .add_system(update_inventory_items)
        .run();
}

#[derive(Component, Debug)]
pub struct Player;

pub fn spawn_player(
    mut commands: Commands,
) {
    commands.spawn(
        Player {},
    );
}

pub fn update_inventory_items(
    player_query: Query<&Player>,
) {
    println!("items updated called!");
    for player in player_query.iter() {
        println!("entered player loop: {:?}", player);
    }
}
```

The terminal output I'm seeing is as follows:
```
2023-08-24T14:48:46.486456Z  INFO bevy_winit::system: Creating new window "Bevy App" (0
v0)
2023-08-24T14:48:46.763620Z  INFO bevy_render::renderer: AdapterInfo { name: "Intel(R)
HD Graphics", vendor: 32902, device: 8880, device_type: IntegratedGpu, driver: "", driv
er_info: "", backend: Dx12 }
items updated called!
entered player loop: Player
items updated called!
entered player loop: Player
items updated called!
entered player loop: Player
items updated called!
entered player loop: Player
items updated called!
entered player loop: Player
items updated called!
entered player loop: Player
...
```

Cargo.toml:
```
[package]
name = "my_bevy_game"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
bevy = { version = "0.10.0", features = ["dynamic_linking"] }
rand = "0.8.5"

# Enable a small amount of optimization in debug mode
[profile.dev]
opt-level = 1

# Enable high optimizations for dependencies (incl. Bevy), but not for our code:
[profile.dev.package."*"]
opt-level = 3
```

**Environment:**
Windows 10.0.1 21H2 9044.3086
LENOVO_MT_80SG_BU_idea_FM_MIIX 310-10ICR
Intel(R) Atom(TM) x5-Z8350 CPU @ 1.44GHz, 1441 MHz, 4 cores
Intel(R) HD Graphics PCI\VEN_8086&DEV_22B0&SUBSYS_380917AA&REV_36\3&11583659&0&10
Driver: C:\WINDOWS\SYSTEM32\DRIVERS\IGDKMD64LP.SYS
Toolchain: stable-x86_64-pc-windows-gnu (default) rustc 1.70.0 (90c541806 2023-05-31)
Bevy 0.10.1

**Expected Behavior:**
I was under the impression that Bevy's query system would only give me results to iterate over when there are actual changes to the components being queried. However, in this case, the component is returned every frame when the system is called, even though the Player component isn't being modified or changed.

Also I noticed high CPU usage even when Bevy was fully idle.

 * Is this the expected behavior of the Bevy query system, or am I missing something?

 * Are there any additional steps I should take to troubleshoot this issue?

Thank you for your time and assistance.

BTW bevy 0.11 does not render any sprites so I am stuck with 0.10.1 as stated here: https://github.com/bevyengine/bevy/issues/8395#issuecomment-1631227183

**Comment by rparrett** [comment, marked as answer]
> I was under the impression that Bevy's query system would only give me results to iterate over when there are actual changes to the components being queried. However, in this case, the component is returned every frame when the system is called, even though the Player component isn't being modified or changed.

That's not the case. To query for only entities with changed components, you would use the [`Changed`](https://docs.rs/bevy/latest/bevy/ecs/query/struct.Changed.html) query filter for the particular component.
